### ISSUE TITLE
How to make web.config based configurable Auditing?

### ISSUE THREAD

**Issue by pankaj2joshi** [thepirat000/Audit.NET#115]
Hello Support,

I'm using Audit.NET for my Web API project, everything is working fine (file based and SQL server database). Now I want to enable a switch by which I can make a key in web.config that decide whether to record event or not? 

So if the flag is false then I'm not initializing all parameters of Audit.net in global.asax.cs. This is working fine in development environment, but when I deploy this on IIS, I'm getting access denied error as mentioned below...

> **"Access to the path 'C:\\Windows\\SysWOW64\\inetsrv\\20180419194955_636597443952597356.json' is denied."**

That means it is creating file even configuration is not done (might be due to attribute above the method). Can you please guide me if such feature is already available for audit.net web API by which I can enable disable auditing by on switch?

**Comment by thepirat000** [comment]
Yes, the problem is that the Filter is applied as an attribute and the library defaults to the `FileDataProvider`. You have some options to ignore the audits by config:

1- Add a custom action on your startup logic that discards the scopes as soon as they are created:

```c#
if (Config.AuditDisabled)
{
    Audit.Core.Configuration.AddCustomAction(ActionType.OnScopeCreated, scope =>
    {
        scope.Discard();
    });
}
```

2- Use a NULL data provider:

```c#
if (Config.AuditDisabled)
{
    Audit.Core.Configuration.Setup().UseDynamicProvider(_ => { });
}
```

**Comment by thepirat000** [comment]
Hi again, please note I've added a global switch to disable the audit logs. This is included starting on version 12.0.0.

You can check the documentation [here](https://github.com/thepirat000/Audit.NET#global-switch-off). You just set the static property `AuditDisabled` to `true`:

```c#
Audit.Core.Configuration.AuditDisabled = true;
```

**Comment by thepirat000** [comment]
Hi again :)

Note that starting on version 12.1.0, you can now setup a convenient global filter that allows configuration per request, as an alternative of decorating the controllers.

If you're interested, take a look at the documentation [here](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.WebApi/README.md#2--global-action-filter).

So now you have a third option to disable the audit as follows:

```c#
public static class WebApiConfig
{
	public static void Register(HttpConfiguration config)
	{
		config.Filters.Add(new AuditApiGlobalFilter(config => config
			.LogActionIf(_ => !YourConfig.IsAuditDisabled && _.ControllerName == "Orders")));
                // log actions on controller Orders iif your audit-disabled configuration is not set
	}
```
